server/middleware/filterTrips.js

```diff
@@ -0,0 +1,21 @@
+import jwt from 'jsonwebtoken';
+import resPonse from '../helpers/responses/response';
+import Trip from '../models/trips';
+
+const trip = new Trip();
+
+const allTrips = async (req, res, next) => {
+  const foundtrips = await trip.getAllTrips();
+  const tripData = foundtrips.rows;
+  const notadmin = jwt.decode(req.headers.authorization);
+  const activeTrips = tripData.filter(x => x.status === 'active');
+  if (!req.headers.authorization || req.headers.authorization === ''
+  || notadmin.isadmin === false || notadmin === null){
```

Comment:
That reusable function I talked about could just be called here as well. Try to make you're done as DRY as possible.